File: apps/examples/tenant-xpto/food-market-05-service/src/domain/food-trading.event.ts
export class FoodTradingEvent {
  eventId: string;
  eventType: string;
  tradingId: string;
  marketId: string;
  itemName: string;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
  traderName: string;
  status: string;
  timestamp: string;

  constructor(partial: Partial<FoodTradingEvent>) {
    Object.assign(this, partial);
    if (!this.eventId) {
      this.eventId = Math.random().toString(36).substring(2, 15);
    }
    if (!this.eventType) {
      this.eventType = 'FOOD_TRADING_CREATED';
    }
    if (!this.timestamp) {
      this.timestamp = new Date().toISOString();
    }
  }
}

File: apps/examples/tenant-xpto/food-market-06-service/src/Domain/Events/FoodTradingEvent.cs
namespace Darueira.FoodMarket.Service06.Domain.Events;

public class FoodTradingEvent
{
    public string EventId { get; set; } = Guid.NewGuid().ToString();
    public string EventType { get; set; } = "FOOD_TRADING_CREATED";
    public string TradingId { get; set; } = string.Empty;
    public string MarketId { get; set; } = string.Empty;
    public string ItemName { get; set; } = string.Empty;
    public decimal Quantity { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal TotalPrice { get; set; }
    public string TraderName { get; set; } = string.Empty;
    public string Status { get; set; } = "CONFIRMED";
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;
}
